### DISCUSSION TITLE
Way to access assets stored in a password-protected server folder?

### DISCUSSION THREAD

**Discussion by vortice3D** [google/model-viewer#2030] [General]
Hi there;

I have a project that requires the 3D assets (GLB/USDZ) to be stored in a password protected web-server folder.

This said, when I access the web page, an authentication dialog ask me for my credentials. After granting access, the model is shown in the webgl context without problems, but when I try to access the AR experience (by means of scene-viewer or quick-look ar modes), as the assets must be downloaded again, an error is thrown.

How must I tackle this inconvenience?

Thanks for your time.

**Comment by shamsail** [comment]
why is it downloading the assets again? are you redirecting the url?

**Reply by vortice3D** [reply]
Hi @shamsail:

Not redirecting any URLs. Sadly, that is the behavior to be expected.

Keep in mind that the **Scene-Viewer** (internal **ARCore** functionality) and **Quick-Look** (internal **ARKit** Functionality) must access again the asset. That is a completely different AR session what is launched, as they are (native) separated processes from the one accessible from Web via WebXR (not usable at the moment in real-world implementations due to a lack of compatible browsers and devices).

Please pay attention to the following paragraph, extracted from the [model-viewer examples page](https://modelviewer.dev/examples/augmentedreality/#ar):
"_Only through WebXR, now our default AR mode, can this be achieved, as the AR session is still inside of the browser. This also removes the need to redownload the model._ "

Best regards.

**Reply by shamsail** [reply]
Ahh, got it, then you can do one thing, hope it will work.

1) get file from the asset path and make a new file locally.

```
let file = await fetch("url/model.glb")
     .then((r) => r.blob())
     .then((blobFile) => new File([blobFile], "model.glb"));
```

2) create a url for local saved file
`URL.createObjectURL(file)`

3) Assign this url to modelViewer.src
`modelViewer.src = URL.createObjectURL(file)`
     
 May be then you will not need password again for AR mode. (during changing the session as you described)
